sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.cs

```diff
@@ -175,28 +200,17 @@ public void Dispose()
             _originalDocument.Dispose();
         }
 
-        internal MutableJsonDocument(JsonDocument document, JsonSerializerOptions? serializerOptions) : this(document, GetBytesFromDocument(document), serializerOptions)
+        private MutableJsonDocument(JsonDocument document, JsonSerializerOptions? serializerOptions) : this(document, null, serializerOptions)
         {
         }
 
-        internal MutableJsonDocument(JsonDocument document, ReadOnlyMemory<byte> utf8Json, JsonSerializerOptions? serializerOptions)
+        private MutableJsonDocument(JsonDocument document, ReadOnlyMemory<byte> utf8Json, JsonSerializerOptions? serializerOptions)
         {
             _original = utf8Json;
             _originalDocument = document;
             _serializerOptions = serializerOptions ?? new JsonSerializerOptions();
```

Comment:
Can we cache the default value?

Reply:
We modify them in places, so I'd rather look at this separately.